### PR TITLE
[KYUUBI #1698] Add kyuubi server operation state rate

### PR DESCRIPTION
**Pull request by zhenjiaguo**
### _Why are the changes needed?_

Add kyuubi server operation state rate. It will be helpful for us to look into kyuubi server running states.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [x] Add screenshots for manual tests if appropriate
Screenshot from 2022-01-19 19-13-21

- [x] [Run test](https://kyuubi.readthedocs.io/en/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: kyuubi-metrics/src/main/scala/org/apache/kyuubi/metrics/MetricsConstants.scala [hunk 1]
```diff
@@ -39,6 +39,7 @@ object MetricsConstants {
   final val OPERATION_OPEN: String = OPERATION + "opened"
   final val OPERATION_FAIL: String = OPERATION + "failed"
   final val OPERATION_TOTAL: String = OPERATION + "total"
+  final val OPERATION_STATE_RATE: String = OPERATION + "state_rate"
```
Comment (yaooqinn) [inline kyuubi-metrics/src/main/scala/org/apache/kyuubi/metrics/MetricsConstants.scala:42]: is rate in a rate redundant？cc @pan3793

File: kyuubi-server/src/main/scala/org/apache/kyuubi/operation/KyuubiOperation.scala [hunk 2]
```diff
@@ -158,4 +160,10 @@ abstract class KyuubiOperation(opType: OperationType, session: Session)
   }
 
   override def shouldRunAsync: Boolean = false
+
+  override def setState(newState: OperationState): Unit = {
+    super.setState(newState)
+    MetricsSystem.tracing(
+      _.markMeter(MetricRegistry.name(OPERATION_STATE_RATE, newState.toString.toLowerCase)))
```
Comment (yaooqinn) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/operation/KyuubiOperation.scala:167]: do we need toLowerCase? seems we are not doing this for OperationTypes

Comment (zhenjiaguo) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/operation/KyuubiOperation.scala:167]: Yes, we need. `OpertationType` is also use `toLowerCase`. Use low case to make them consistent.

File: kyuubi-metrics/src/main/scala/org/apache/kyuubi/metrics/MetricsConstants.scala (hunk 1 above)
Comment (zhenjiaguo) [reply kyuubi-metrics/src/main/scala/org/apache/kyuubi/metrics/MetricsConstants.scala:42]: Ok, I will remove it.

**Review by yaooqinn — APPROVED** [review]
LGTM

**Comment by yaooqinn** [comment]
thanks, merged to master

**Comment by yaooqinn** [comment]
thanks, merged to master and v1.4.1

**Pull request closed without merging**